src/FromFile.jl

```diff
@@ -12,15 +13,44 @@ function __init__()
     @require Revise="295af30f-e4ad-537b-8983-00126c2a3abe" track(mod, path) = Revise.track(mod, path)
 end
 
-macro from(path::String, ex::Expr)
+macro from(path, ex::Expr)
     esc(from_m(__module__, __source__, path, ex))
 end
 
-macro from(path::String)
+macro from(path)
     esc(from_m(__module__, __source__, path, Expr(:block))) 
 end
 
-function from_m(m::Module, s::LineNumberNode, path::String, root_ex::Expr)
+# NOTE: we'd like to keep this pacakge minimal deps
+# so just copy paste this one-liner
+# Source: IsUrl.jl (MIT license)
+function isurl(str::AbstractString)
+    windowsregex = r"^[a-zA-Z]:[\\]"
+    urlregex = r"^[a-zA-Z][a-zA-Z\d+\-.]*:"
+    return !occursin(windowsregex, str) && occursin(urlregex, str)
+end
+
+function from_m(m::Module, s::LineNumberNode, path_or_url, root_ex::Expr)
+    path_or_url = eval_path_const(m, path_or_url)
+    if isurl(path_or_url)
+        from_remote_file(m, s, path_or_url, root_ex)
+    else
+        from_local_file(m, s, path_or_url, root_ex)
+    end
+end
+
+function eval_path_const(m::Module, path)
```

Comment:
Maybe assert that this returns a string? The `eval` here could do almost anything (and change the state of `m` whilst doing so), which feels a bit damgerous around typos to me, so I'd like to add at least one guardrail.

Reply:
hmm, I initially only allow string interpolation, but then I realize things like `joinpath` also make sense as the `path` which was why I allow arbitrary Julia expression in the end. Besides, this is gonna be the user's own code so they are at their own risk since they are already importing a random script from somewhere (either locally or remotely) we can't do much for this.

also note, if we allow string interpolation, it will also allow arbitrary code via side effects, e.g

```julia
function my_evil_code_that_returns_string()
    push!(DEPOT_PATH, "my_evil_depot_path")
    return "jaja"
end

@from "$(my_evil_code_that_returns_string())" import X as Y
```

Reply:
I don't mean as a security risk,  just as a potential footgun from typoing the wrong function.

Reply:
I see what you mean! good point, let's only allow string/string expr then!